packages/react/src/reactrouterv6.tsx

```diff
@@ -0,0 +1,162 @@
+import { Transaction, TransactionContext } from '@sentry/types';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+import React from 'react';
+
+import { Action, Location } from './types';
+
+interface RouteObject {
+  caseSensitive?: boolean;
+  children?: RouteObject[];
+  element?: React.ReactNode;
+  index?: boolean;
+  path?: string;
+}
+
+type Params<Key extends string = string> = {
+  readonly [key in Key]: string | undefined;
+};
+
+interface RouteMatch<ParamKey extends string = string> {
+  params: Params<ParamKey>;
+  pathname: string;
+  route: RouteObject;
+}
+
+type UseEffect = (cb: () => void, deps: unknown[]) => void;
+type UseLocation = () => Location;
+type UseNavigationType = () => Action;
+type CreateRoutesFromChildren = (children: JSX.Element[]) => RouteObject[];
+type MatchRoutes = (routes: RouteObject[], location: Location) => RouteMatch[] | null;
+
+let _useEffect: UseEffect;
+let _useLocation: UseLocation;
+let _useNavigationType: UseNavigationType;
+let _createRoutesFromChildren: CreateRoutesFromChildren;
+let _matchRoutes: MatchRoutes;
+let _customStartTransaction: (context: TransactionContext) => Transaction | undefined;
+let _startTransactionOnLocationChange: boolean;
+let _startTransactionOnPageLoad: boolean;
+
+export function reactRouterV6Instrumentation(
+  useEffect: UseEffect,
+  useLocation: UseLocation,
+  useNavigationType: UseNavigationType,
+  createRoutesFromChildren: CreateRoutesFromChildren,
+  matchRoutes: MatchRoutes,
+) {
+  return (
+    customStartTransaction: (context: TransactionContext) => Transaction | undefined,
+    startTransactionOnPageLoad = true,
+    startTransactionOnLocationChange = true,
+  ): void => {
+    _useEffect = useEffect;
```

Comment:
Can we start the transaction here and just update the transaction name in the useeffect call?

Also, is there a way we can listen directly onto `history` instead of using `useEffect`?

Reply:
Updated to start pageload transaction on init 👍 

Tried to find a simple solution to access `history` but it doesn't look possible without risking reliability (as they're encouraging to use `useNavigate` and `useLocation` hooks in v6, so even if we try to use global history, they may conflict?). Combining them with `useEffect` seems to be the most reliable, but can dig deeper if you like.

Reply:
Ok - no worries, let's keep using the hooks then!